subprojects/core/src/main/java/org/gradle/execution/TaskNameResolvingBuildTaskScheduler.java

```diff
@@ -54,6 +57,28 @@ public void scheduleRequestedTasks(GradleInternal gradle, @Nullable EntryTaskSel
                 plan.addEntryTasks(taskSelection.getTasks());
             }
         }
+        validateCompatibleTasksRequested(plan);
+    }
+
+    /**
+     * Validates the tasks to be run are mutually compatible.
+     * <p>
+     * Currently, this checks that {@code init} is not run along with any other tasks.
+     *
+     * @param plan execution plan containing requested tasks to validate
+     */
+    private void validateCompatibleTasksRequested(ExecutionPlan plan) {
+        //noinspection ConstantValue support mocking in tests
+        if (null != plan.getContents()) {
+            Set<Task> requestedTasks = plan.getContents().getRequestedTasks();
+            if (requestedTasks.size() > 1 && requestedTasks.stream().anyMatch(t -> t.getName().equals("init"))) { // TODO: Consider moving the InitBuiltInCommand (and help) to core, as they are not Software Platform-specific
```

Comment:
Maybe we can inject `List<BuiltInCommand>` and check if the requested task is a built-in command that requires exclusive access?

Reply:
I just learned injecting a list like this was possible from another PR earlier today.  This is a good idea, I'll make the change.